sklearn/model_selection/_validation.py

```diff
@@ -505,6 +506,21 @@ def _fit_and_score(estimator, X, y, scorer, train, test, verbose,
         end_msg = "%s, total=%s" % (msg, logger.short_format_time(total_time))
         print("[CV] %s %s" % ((64 - len(end_msg)) * '.', end_msg))
 
+    # check for inf or -inf values in the train and test scores   
+    if(np.any(np.logical_not(np.isfinite(a))) for a in test_scores.values() ):
```

Comment:
Please don't put parentheses around the whole condition

Reply:
I also think the logic here is not quite right.

Reply:
`test_scores` will only contain numbers, nan and inf values, right? `np.isfinite()` detects `nan`, `inf` and `-inf` values as non-finite. Other than the detection of `nan` here, is there some other problem?